funding/manager.go

```diff
@@ -1039,13 +1056,83 @@ func (f *Manager) stateStep(channel *channeldb.OpenChannel,
 	return fmt.Errorf("undefined channelState: %v", channelState)
 }
 
+// waitForFundingSpend waits for any of the inputs to the provided funding
+// transaction to be spent, and returns the txid of the spending transaction.
+// This is used to detect the case where the funding tx is double spent and
+// rendered invalid.
+func (f *Manager) waitForFundingSpend(fundingTx *wire.MsgTx,
+	fndBroadcastHeight uint32) (*chainhash.Hash, error) {
+	errs := make(chan error)
+	spends := make(chan *chainntnfs.SpendDetail)
+
+	for _, txIn := range fundingTx.TxIn {
+
+		f.wg.Add(1)
+		go func(txIn *wire.TxIn) {
```

Comment:
If only one of the funding txn's input is double-spent, the remaining goroutines will linger instead of being cleaned up

Reply:
Added a done channel which closes on one of the channels getting a spend notificatoin